src/token-gated-minter/TokenGatedMinter.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {IERC165} from "forge-std/interfaces/IERC165.sol";
+import {IERC721Drop} from "zora-drops-contracts/interfaces/IERC721Drop.sol";
+import {ITokenBalance} from "../utils/ITokenBalance.sol";
+
+contract TokenGatedMinter {
+    struct TokenGate {
+        uint256 amount;
+        uint256 mintPrice;
+        uint256 mintLimit;
+    }
+
+    // token gate contract address => token gate details
+    mapping(address => TokenGate) public tokenGates;
+
+    // minter address => token gate contract address => has minted
+    mapping(address => mapping(address => bool)) public hasMintedWithTokenGate;
+
+    address payable public immutable mintingToken;
+
+    modifier onlyTokenAdmin() {
+        require(
```

Comment:
I’m fine with require here if it’s used consistently. Especially because this may have no ui to start.